### PR TITLE
More helpful errors

### PR DESCRIPTION
**Pull request by dmdashenkov**
This PR improves the errors a user may see when building a project with Model Compiler.

### CODE REVIEW

**Comment by dmdashenkov** [comment]
@armiol, PTAL.

File: base/src/main/java/io/spine/code/proto/RejectionsFile.java [hunk 1]
```diff
@@ -42,16 +53,34 @@ private RejectionsFile(FileDescriptor descriptor) {
...
         return result;
     }
 
+    private static void checkMatchesConvention(SourceFile file) {
+        FileDescriptor descriptor = file.descriptor();
+        checkArgument(isRejections(descriptor),
+                      "`%s`. A rejection file must have a name ending in `rejections.proto`.",
+                      file);
+        checkArgument(!descriptor.getOptions().getJavaMultipleFiles(),
+                      "`%s`. A rejection file should generate Java classes into a single file. " +
```
Comment (armiol) [inline base/src/main/java/io/spine/code/proto/RejectionsFile.java]:
```
Rejection Java classes must be generated into a single file.  Please set `java_multiple_files` to `false`.
```

File: base/src/main/java/io/spine/code/proto/RejectionsFile.java [hunk 2]
```diff
@@ -42,16 +53,34 @@ private RejectionsFile(FileDescriptor descriptor) {
...
+                              "Please set `java_multiple_files` to `false`.",
+                      file);
+        Optional<SimpleClassName> outerClass = SimpleClassName.declaredOuterClassName(descriptor);
+        outerClass.ifPresent(name -> checkArgument(
+                isValidOuterClassName(name),
+                "%s. A rejection file must have the `java_outer_classname` ending in " +
+                        "`Rejections` or have none at all.",
```
Comment (armiol) [inline base/src/main/java/io/spine/code/proto/RejectionsFile.java]: `...or have no ending at all`.

File: base/src/main/java/io/spine/code/proto/SourceFile.java [hunk 3]
```diff
@@ -68,16 +68,18 @@ private static Path toPath(FileDescriptor file) {
...
-     * <p>A valid rejections file must have:
+     * <p>A valid rejections file must:
      * <ul>
-     *     <li>The file name which ends on
-     *         {@link io.spine.base.MessageFile#REJECTIONS “rejections.proto”}.
-     *     <li>The option {@code java_multiple_files} set to {@code false}.
-     *     <li>
+     *     <li>be named ending on {@link io.spine.base.MessageFile#REJECTIONS “rejections.proto”};
+     *     <li>have the {@code java_multiple_files} option set to {@code false};
+     *     <li>not have the option {@code java_outer_classname} or have a value which ends with
```
Comment (armiol) [inline base/src/main/java/io/spine/code/proto/SourceFile.java]: Same as above.

**Pull request merged**